tpcp/_dataset.py

```diff
@@ -85,38 +86,75 @@ def _create_check_index(self):
         return index_1
 
     @property
-    def groups(self) -> List[Union[str, Tuple[str, ...]]]:
-        """Get all groups based on the set groupby level.
+    def group_labels(self) -> List[Tuple[str, ...]]:
+        """Get all group labels of the dataset based on the set groupby level.
 
-        This will either return a list of strings/integers, if there is only a single group level or index column.
-        If there are multiple groupy levels/index columns, it will return a list of named tuples.
+        This will return a list of named tuples.
+        The tuples will contain only one entry if there is only one groupby level or index column.
+
+        The elements of the named tuples will have the same names as the groupby columns and will be in the same order.
 
         Note, that if one of the groupby levels/index columns is not a valid Python attribute name (e.g. in contains
         spaces or starts with a number), the named tuple will not contain the correct column name!
         For more information see the documentation of the `rename` parameter of :func:`collections.namedtuple`.
+
+        For some examples and additional explanation see this :ref:`example <custom_dataset_basics>`.
         """
-        if len(self._get_groupby_columns()) == 1:
-            return self._get_unique_groups().to_list()
-        return list(self._get_unique_groups().to_frame().itertuples(index=False, name=type(self).__name__))
+        return list(
+            self._get_unique_groups().to_frame().itertuples(index=False, name=type(self).__name__ + "GroupLabel")
+        )
 
     @property
-    def group(self) -> Union[str, Tuple[str, ...]]:
-        """Get the current group.
+    def groups(self) -> List[Tuple[str, ...]]:
+        warnings.warn(
+            "The attribute `groups` is deprecated and will be removed in a future version. "
+            "Use `group_labels` instead.",
+            DeprecationWarning,
+        )
+        return self.group_labels
+
+    @property
+    def group_label(self) -> Tuple[str, ...]:
+        """Get the current group label.
 
         Note, this attribute can only be used, if there is just a single group.
-        If there is only a single groupby column or column in the index, this will return a string.
-        Otherwise, this will return a named tuple.
+        This will return a named tuple. The tuple will contain only one entry if there is only a single groupby column
+        or column in the index.
         """
         self.assert_is_single_group("group")
-        return self.groups[0]
+        return self.group_labels[0]
+
+    @property
+    def group(self) -> Tuple[str, ...]:
```

Comment:
I think the deprectaed methods need a docstring for the linter. But a one-liner should be fine